x-pack/plugin/inference/src/test/java/org/elasticsearch/cluster/metadata/SemanticTextClusterMetadataTests.java

```diff
@@ -35,7 +33,10 @@ public void testCreateIndexWithSemanticTextField() {
             "test",
             client().admin().indices().prepareCreate("test").setMapping("field", "type=semantic_text,model_id=test_model")
         );
-        assertEquals(Map.of("test_model", Set.of("field")), indexService.getMetadata().getFieldsForModels());
+        assertEquals(
+            indexService.getMetadata().getFieldInferenceMetadata().getFieldInferenceOptions().get("field").inferenceId(),
+            "test_model"
```

Comment:
Nitpick: the `assertEquals` method signature is `assertEquals(<expected>, <actual>)`. Reversing the args is the same functionally, but if the assertion fails it will generate a message with an incorrect expected value.

Reply:
Well spotted! I'm changing that to Hamcrest assertions, which are easier to read 👍